main.go

```diff
@@ -1,51 +1,135 @@
+// iterscraper scrapes information from a website where URLs contain an incrementing integer.
+// Information is retrieved from HTML5 elements, and outputted as a CSV.
 package main
 
 import (
+	"encoding/csv"
 	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
 	"sync"
-)
 
-var (
-	urlBase       string
-	idLow, idHigh int
-	concurrency   int
-	outfile       string
-	nameQuery     string
-	addressQuery  string
-	phoneQuery    string
-	emailQuery    string
+	"github.com/PuerkitoBio/goquery"
 )
 
 func main() {
-	// Get flags
-	flag.StringVar(&urlBase, "url", "http://example.com/v/%d", `The URL you wish to scrape, containing "%d" where the id should be substituted`)
-	flag.IntVar(&idLow, "from", 0, "The first ID that should be searched in the URL - inclusive.")
-	flag.IntVar(&idHigh, "to", 1, "The last ID that should be searched in the URL - exclusive")
-	flag.IntVar(&concurrency, "concurrency", 1, "How many scrapers to run in parallel. (More scrapers are faster, but more prone to rate limiting or bandwith issues)")
-	flag.StringVar(&outfile, "output", "output.csv", "Filename to export the CSV results")
-	flag.StringVar(&nameQuery, "nameQuery", ".name", "JQuery-style query for the name element")
-	flag.StringVar(&addressQuery, "addressQuery", ".address", "JQuery-style query for the address element")
-	flag.StringVar(&phoneQuery, "phoneQuery", ".phone", "JQuery-style query for the phone element")
-	flag.StringVar(&emailQuery, "emailQuery", ".email", "JQuery-style query for the email element")
-
+	var (
+		urlTemplate = flag.String("url", "http://example.com/v/%d", "The URL you wish to scrape, containing \"%d\" where the id should be substituted")
+		idLow       = flag.Int("from", 0, "The first ID that should be searched in the URL - inclusive.")
+		idHigh      = flag.Int("to", 1, "The last ID that should be searched in the URL - exclusive")
+		concurrency = flag.Int("concurrency", 1, "How many scrapers to run in parallel. (More scrapers are faster, but more prone to rate limiting or bandwith issues)")
+		outfile     = flag.String("output", "output.csv", "Filename to export the CSV results")
+		name        = flag.String("nameQuery", ".name", "JQuery-style query for the name element")
+		address     = flag.String("addressQuery", ".address", "JQuery-style query for the address element")
+		phone       = flag.String("phoneQuery", ".phone", "JQuery-style query for the phone element")
+		email       = flag.String("emailQuery", ".email", "JQuery-style query for the email element")
+	)
 	flag.Parse()
 
-	// Use waitgroup so we can keep track of tasks
+	columns := []string{*name, *address, *phone, *email}
+	headers := []string{"name", "address", "phone", "email"}
+	// url and id are added as the first two rows.
```

Comment:
I would call it columns, because those are the column names ... but it's really up to you